### PR TITLE
v3.1.7 commits and release notes

### PR DESCRIPTION
**Pull request by cary-ilm**
Commits and notes for v3.1.7, just the recent ARMv7/NEON fixes, and the zib fix.

@Vertexwahn, I'm unclear on which of the bazel PR's should go into the patch releases. If you can identify ones that should go in, I'm happy to merge them as well.

### CODE REVIEW

File: cmake/OpenEXRSetup.cmake [hunk 1]
```diff
@@ -224,6 +224,16 @@ if(OPENEXR_FORCE_INTERNAL_ZLIB OR NOT TARGET ZLIB::ZLIB)
     set(zlibpostfix "d")
   endif()
 
+  add_library(zlib_static STATIC IMPORTED GLOBAL)
```
Comment (meshula) [inline cmake/OpenEXRSetup.cmake:227]: is this chunk a whitespace change? I'm not spotting the diff.

**Review by meshula — APPROVED** [review]
lgtm, as you noted Bazel patches might go along with it.

File: cmake/OpenEXRSetup.cmake (hunk 1 above)
Comment (cary-ilm) [reply cmake/OpenEXRSetup.cmake:227]: Not sure I follow, lines 227-236 are new, right?

Comment (meshula) [reply cmake/OpenEXRSetup.cmake:227]: I think the diff viewer messed up. I added the comment on a chunk that I don't see in the diff now.

**Comment by Vertexwahn** [comment]
@cary-ilm Please cherry-pick this commit https://github.com/AcademySoftwareFoundation/openexr/pull/1370

**Comment by Vertexwahn** [comment]
@cary-ilm I manually stashed together the Bazel Contributions to OpenEXR 3.1.7 [here](https://github.com/AcademySoftwareFoundation/openexr/pull/1373) on top of this PR. You can cherry-pick commit 843ed9153b081309f589a32e2b0dde8d5a222b4b to this PR to get all relevant Bazel build contributions for OpenEXR 3.1.7.

**Comment by cary-ilm** [comment]
@Vertexwahn, I merged the commit from #1373 in, thanks.

**Pull request merged**